File: app/src/main/java/com/iseasoft/iseaiptv/ui/fragment/PlaylistFragment.java
package com.iseasoft.iseaiptv.ui.fragment;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.iseasoft.iseaiptv.R;
import com.iseasoft.iseaiptv.adapters.PlaylistAdapter;
import com.iseasoft.iseaiptv.helpers.Router;
import com.iseasoft.iseaiptv.listeners.OnPlaylistListener;
import com.iseasoft.iseaiptv.models.Playlist;
import com.iseasoft.iseaiptv.utils.PreferencesUtility;
import com.iseasoft.iseaiptv.utils.Utils;

import java.util.ArrayList;
import java.util.Collections;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

/**
 * A simple {@link Fragment} subclass.
 */
public class PlaylistFragment extends Fragment {

    Unbinder unbinder;
    @BindView(R.id.list)
    RecyclerView list;
    @BindView(R.id.empty_container)
    LinearLayout emptyContainer;

    private PlaylistAdapter playlistAdapter;

    public static PlaylistFragment newInstance() {
        PlaylistFragment fragment = new PlaylistFragment();
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_playlist, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Utils.modifyListViewForVertical(getContext(), list);
        loadPlaylist();
    }

    private void loadPlaylist() {
        ArrayList<Playlist> playlists = PreferencesUtility.getInstance(getActivity()).getPlaylist();
        if (playlists == null || playlists.size() == 0) {
            emptyContainer.setVisibility(View.VISIBLE);
            list.setVisibility(View.GONE);
        } else {
            emptyContainer.setVisibility(View.GONE);
            list.setVisibility(View.VISIBLE);
            Collections.reverse(playlists);
            playlistAdapter = new PlaylistAdapter(playlists, new OnPlaylistListener() {
                @Override
                public void onPlaylistItemClicked(Playlist item) {
                    PreferencesUtility.getInstance(getActivity()).savePlaylist(item);
                    Router.navigateToMainScreen(getActivity(), true);
                }
            });
            list.setAdapter(playlistAdapter);
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }

    @OnClick(R.id.add_playlist)
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.add_playlist:
                getActivity().openOptionsMenu();
                break;
        }
    }


}

File: app/src/main/java/com/iseasoft/iseaiptv/ui/fragment/PlaylistFragment.kt
package com.iseasoft.iseaiptv.ui.fragment


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import butterknife.Unbinder
import com.iseasoft.iseaiptv.R
import com.iseasoft.iseaiptv.adapters.PlaylistAdapter
import com.iseasoft.iseaiptv.helpers.Router
import com.iseasoft.iseaiptv.listeners.OnPlaylistListener
import com.iseasoft.iseaiptv.models.Playlist
import com.iseasoft.iseaiptv.utils.PreferencesUtility
import com.iseasoft.iseaiptv.utils.Utils
import java.util.*

/**
 * A simple [Fragment] subclass.
 */
class PlaylistFragment : Fragment() {

    internal lateinit var unbinder: Unbinder

    @BindView(R.id.list)
    @JvmField
    var list: RecyclerView? = null

    @BindView(R.id.empty_container)
    @JvmField
    var emptyContainer: LinearLayout? = null

    private var playlistAdapter: PlaylistAdapter? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_playlist, container, false)
        unbinder = ButterKnife.bind(this, view)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        context?.let { list?.let { it1 -> Utils.modifyListViewForVertical(it, it1) } }
        loadPlaylist()
    }

    private fun loadPlaylist() {
        val playlists = PreferencesUtility.getInstance(activity).playlist
        if (playlists == null || playlists.size == 0) {
            emptyContainer!!.visibility = View.VISIBLE
            list!!.visibility = View.GONE
        } else {
            emptyContainer!!.visibility = View.GONE
            list!!.visibility = View.VISIBLE
            Collections.reverse(playlists)
            playlistAdapter = PlaylistAdapter(playlists, object : OnPlaylistListener {
                override fun onPlaylistItemClicked(item: Playlist) {
                    PreferencesUtility.getInstance(activity).savePlaylist(item)
                    activity?.let { Router.navigateToMainScreen(it, true) }
                }
            })
            list!!.adapter = playlistAdapter
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        unbinder.unbind()
    }

    @OnClick(R.id.add_playlist)
    fun onViewClicked(view: View) {
        when (view.id) {
            R.id.add_playlist -> activity!!.openOptionsMenu()
        }
    }

    companion object {

        fun newInstance(): PlaylistFragment {
            return PlaylistFragment()
        }
    }


}
